### PR TITLE
Add ScrollController

### PR DESCRIPTION
**Pull request by abarth**
If you pass a ScrollController to a Scrollable2, you can use the controller to
read and write the scroll offset without having to find the Scrollable2State
object.

### CODE REVIEW

**Comment by abarth** [comment]
@Hixie

File: packages/flutter/lib/src/widgets/scroll_controller.dart [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+  ///
+  /// Immediately after the jump, a ballistic activity is started, in case the
+  /// value was out of range.
+  void jumpTo(double value) {
+    assert(_positions.isNotEmpty, 'ScrollController not attached to any scroll views.');
+    assert(_positions.length == 1, 'ScrollController attached to multiple scroll views.');
+     _positions.single.jumpTo(value);
+  }
+
+  /// Registesr the given position with this controller.
```
Comment (Hixie) [inline packages/flutter/lib/src/widgets/scroll_controller.dart]: typo

File: packages/flutter/lib/src/widgets/scrollable.dart [hunk 2]
```diff
@@ -125,14 +129,16 @@ class Scrollable2State extends State<Scrollable2> with TickerProviderStateMixin
     if (config.physics != null)
       physics = config.physics.applyTo(physics);
     final ScrollPosition oldPosition = position;
-    _position = physics.createScrollPosition(physics, this, oldPosition);
-    assert(position != null);
     if (oldPosition != null) {
+      config.controller?.detach(oldPosition);
       // It's important that we not do this until after the viewport has had a
```
Comment (Hixie) [inline packages/flutter/lib/src/widgets/scrollable.dart]: while you're here can you clarify my comment here by changing "do this" to "dispose of the old position"? I just got confused reading my own comment...

**Comment by Hixie** [comment]
LGTM

**Pull request merged**